source/validate.cpp

```diff
@@ -332,14 +270,34 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+void ProcessIds(ValidationState_t& _, const spv_parsed_instruction_t& inst) {
+  if (inst.result_id) {
+    _.usedefs().AddDef(
+        {inst.result_id, inst.opcode,
+         std::vector<uint32_t>(inst.words, inst.words + inst.num_words)});
+  }
+  for (auto op = inst.operands; op != inst.operands + inst.num_operands; ++op) {
+    if (op->type == SPV_OPERAND_TYPE_ID)
```

Comment:
There are also SPV_OPERAND_TYPE_TYPE_ID, .... *_MEMORY_SEMANTICS_ID, *_SCOPE_ID.
Maybe we need a utility function bool spvOperandTypeIsID(spv_operand_type_t)